src/core/tritonserver.cc

```diff
@@ -1294,6 +1294,26 @@ TRITONSERVER_InferenceRequestAppendInputData(
   return nullptr;  // Success
 }
 
+TRITONSERVER_Error*
+TRITONSERVER_InferenceRequestAppendInputDataForDevice(
+    TRITONSERVER_InferenceRequest* inference_request, const char* name,
+    const void* base, size_t byte_size, TRITONSERVER_MemoryType memory_type,
+    int64_t memory_type_id, TRITONSERVER_InstanceGroupKind device_kind,
+    int device_id)
+{
+  ni::InferenceRequest* lrequest =
+      reinterpret_cast<ni::InferenceRequest*>(inference_request);
+
+  // Madhu fixme
```

Comment:
@deadeyegoodwin @GuanLuo Should we assert here that the device_kind is GPU and update the documentation to say appendDataForDevice is currently only supported for gpu device ?